javascript/apps/taiga/src/app/shared/comments/components/comment-detail/comment-detail.component.ts

```diff
@@ -0,0 +1,141 @@
+ */
+
+import { CommonModule } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+} from '@angular/core';
+import { TRANSLOCO_SCOPE, TranslocoModule } from '@ngneat/transloco';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { Store } from '@ngrx/store';
+import { RxState } from '@rx-angular/state';
+import { TuiAutoFocusModule } from '@taiga-ui/cdk';
+import {
+  TuiButtonModule,
+  TuiDataListModule,
+  TuiSvgModule,
+} from '@taiga-ui/core';
+import { ShortcutsService } from '@taiga/core';
+import { Permissions, Project, Story, User, UserComment } from '@taiga/data';
+import { Subject, takeUntil } from 'rxjs';
+import { selectUser } from '~/app/modules/auth/data-access/+state/selectors/auth.selectors';
+import { selectCurrentProject } from '~/app/modules/project/data-access/+state/selectors/project.selectors';
+import { StoryDetailActions } from '~/app/modules/project/story-detail/data-access/+state/actions/story-detail.actions';
+import { DropdownModule } from '~/app/shared/dropdown/dropdown.module';
+import { filterNil } from '~/app/shared/utils/operators';
+
+@UntilDestroy()
+@Component({
+  selector: 'tg-comment-detail',
+  standalone: true,
+  templateUrl: './comment-detail.component.html',
+  styleUrls: ['./comment-detail.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  imports: [
+    CommonModule,
+    DropdownModule,
+    TranslocoModule,
+    TuiSvgModule,
+    TuiDataListModule,
+    TuiButtonModule,
+    TuiAutoFocusModule,
+  ],
+  providers: [
+    RxState,
+    {
+      provide: TRANSLOCO_SCOPE,
+      useValue: 'comments',
+    },
+  ],
+})
+export class CommentDetailComponent {
+  @Input({ required: true }) public comment!: UserComment;
+  @Input({ required: true }) public story!: Story | null;
+
+  @Output() public highlightComment = new EventEmitter<string | undefined>();
+
+  public model$ = this.state.select();
+
+  public showDeleteCommentConfirm = false;
+  public canComment!: boolean;
+
+  private unsetDelete$: Subject<void> = new Subject();
+
+  public get commentOptionsState() {
+    return this.#commentOptionsState;
+  }
+
+  public set commentOptionsState(value: boolean) {
+    this.#commentOptionsState = value;
+    if (!value) {
+      this.showDeleteCommentConfirm = false;
+      this.highlightComment.next(undefined);
+      this.unsetDelete$.next();
+    } else {
+      this.setDeleteShortcut();
+    }
+  }
+
+  #commentOptionsState = false;
+
+  constructor(
+    private readonly store: Store,
+    private readonly state: RxState<{
+      user: User;
+      project: Project;
+    }>,
+    private readonly shortcutsService: ShortcutsService
+  ) {
+    this.state.connect('user', this.store.select(selectUser).pipe(filterNil()));
+    this.state.connect(
+      'project',
+      this.store.select(selectCurrentProject).pipe(filterNil())
+    );
+
+    this.canComment = this.state
```

Comment:
`canComment` is an input of tg-comment you don't need the same again.

Al we have this for permissions:

```ts
    this.state.connect(
      'canComment',
      this.hasPermissions.hasPermissions$('story', ['comment'])
    );
```